libraries/AP_IOMCU/iofirmware/iofirmware.cpp

```diff
@@ -243,27 +484,46 @@ void AP_IOMCU_FW::update()
         // turn amber off
         AMBER_SET(0);
     }
-
     // update status page at 20Hz
     if (now - last_status_ms > 50) {
         last_status_ms = now;
         page_status_update();
     }
 
-    // run remaining functions at 1kHz
-    if (now != last_loop_ms) {
-        last_loop_ms = now;
-        heater_update();
+    // run fast loop functions at 1Khz
+    if (now_us - last_fast_loop_us >= 1000)
+    {
+        last_fast_loop_us = now_us;
         rcin_update();
-        safety_update();
-        rcout_mode_update();
         rcin_serial_update();
+    }
+
+    // run remaining functions at 100Hz
+    // these are all relatively expensive and take ~10ms to complete
+    // so there is no way they can effectively be run faster than 100Hz
+    if (now - last_slow_loop_ms > 10) {
+        last_slow_loop_ms = now;
+        heater_update();
```

Comment:
running the heater_update at 100Hz instead of 1kHz will impact the noise seen on the builtin compass. We moved it to be 1kHz to dither the heater to produce less magnetic noise